src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,91 @@
+package parser;
+
+import commands.*;
+import exceptions.DukeException;
+
+public class Parser {
+    /**
+     * Returns a Command class based on input type.
+     * @param input An array of strings split by spaces.
+     * @return Command
+     */
+    public Command parseCommand(String[] input) throws DukeException {
+        StringBuilder name;
+        StringBuilder current;
+        String s1 = input[0];
+        switch (s1) {
+
+        case ListCommand.COMMAND:
+            return new ListCommand();
+
+        case ByeCommand.COMMAND:
+            return new ByeCommand();
+
+        case MarkCommand.COMMAND:
+            if (input.length != 2) {
+                throw new DukeException("Invalid arguments detected! Please enter a index.");
+            }
+            return new MarkCommand(Integer.parseInt(input[1]));
+
+        case UnmarkCommand.COMMAND:
+            if (input.length != 2) {
+                throw new DukeException("Invalid arguments detected! Please enter a index.");
+            }
+            return new UnmarkCommand(Integer.parseInt(input[1]));
+
+        case TodoCommand.COMMAND:
+            name = new StringBuilder();
+            for (int j = 1; j < input.length; j++) {
+                name.append(input[j]).append(" ");
+            }
+            return new TodoCommand(name.toString());
```

Comment:
Would love to hear your rational of why you decided to place the parameters extraction inside the switch statements! 😄

Reply:
Initially, I was also unsure if whether parameters should be extracted within the Parser class, or during the execution phase. However, I decided that the execution phase should only be executed when there's a valid input given by the user; any wrong input would be rejected by the parser. I hope that answers ur question... do lmk if you have any recommendations or any queries. Thx!